Fix upgrade

ss-es opened the pull request.

ss-es committed:
do not load decided upgrade cert if unnecessary

ss-es committed:
add log

ss-es committed:
TEST remove checks for time

ss-es committed:
fix

ss-es committed:
log hotshot config

ss-es committed:
Revert "TEST remove checks for time"

This reverts commit 70d59d6b8f62c082e664dca661b8a3289b56a13d.

ss-es committed:
Revert "fix"

This reverts commit cab9acdf25da660c112c68c52e380b448ff8f4d1.

ss-es committed:
Revert "do not load decided upgrade cert if unnecessary"

This reverts commit 1085f150265e47c6a8cf9d4385ba4af196557072.

ss-es committed:
fix logging

ss-es committed:
add more logging

ss-es committed:
i give up log everything

ss-es committed:
Revert "i give up log everything"

This reverts commit bb5383f2be1c6ce35e36522b251f807462c239c0.

ss-es committed:
use estimate of decide_by block height not view number

bfish713 commented on hotshot-task-impls/src/upgrade.rs:
```diff
@@ -304,8 +309,10 @@ impl<TYPES: NodeType, V: Versions> UpgradeTaskState<TYPES, V> {
-                    let upgrade_finish_epoch =
-                        epoch_from_block_number(new_version_first_view + 10, self.epoch_height);
+                    let upgrade_finish_epoch = epoch_from_block_number(
+                        last_proposal_block + (new_version_first_view - last_proposal_view) + 10,
```
isn't this basically `last_proposal_block + TYPES::UPGRADE_CONSTANTS.finish_offset + 10`

I'm trying to describe what this code is checking and this should be, the block the upgrade finishes on should be in the same epoch it started.  so we basically need current block + how many views it'll take from now to apply + buffer

ss-es replied:
conceptually it's almost the same thing as what you just said, except that I think `last_proposal_block + finish_offset + 10` could be earlier than `new_version_first_view`

ss-es replied:
the way I think of this is just (block height of last proposal) + (the maximum number of blocks we could possibly produce by `new_version_first_view`)

ss-es committed:
add vote check and configure epoch_start_block

ss-es committed:
configure start_proposing_view in the past

ss-es committed:
revert change

bfish713 commented on hotshot-task-impls/src/upgrade.rs:
```diff
@@ -160,6 +160,46 @@ impl<TYPES: NodeType, V: Versions> UpgradeTaskState<TYPES, V> {
                     proposal.data.view_number()
                 );
 
+                let epoch_upgrade_checks = if V::Upgrade::VERSION == V::Epochs::VERSION {
```
Just to confirm this is and should be a copy of the same checks on the proposal side.  Looks the same to me but just wanted a double check

ss-es committed:
revert vote check

imabdulbasit committed:
use correct to_block for stake table events fetching (#3103)

(cherry picked from commit 87302f0cca4dcffa5a226fcbe55ff5065978a11b)

imabdulbasit committed:
skip reward header dependencies for first two epochs

(cherry picked from commit 0babc5aa949c3e350ddaa88fc7561e9a0036ff18)

ss-es committed:
update genesis decaf.toml

github-actions[bot] committed:
[Backport release-fix-upgrade] Fix Synced Block Height Metric in new Version (#3125)

Fix Synced Block Height Metric in new Version (#3108)

fix metrics; add synced block height to new decide logic

(cherry picked from commit eb3170dd39737bfc127e3dde16da0c74c05f8d37)

Co-authored-by: Brendon Fish <bfish713@gmail.com>

github-actions[bot] committed:
[Backport release-fix-upgrade] Remove failed stake table catchup request to allow new catchup request (#3154)

Remove failed stake table catchup request to allow new catchup request (#3130)

remove failed catchup request

(cherry picked from commit f871bb61ba23b9e7362be871fea2f5c2154d5aee)

Co-authored-by: Abdul Basit <45506001+imabdulbasit@users.noreply.github.com>

bfish713 committed:
Safegaurd Endpoint (#3174)

don't trigger catchup for high epoch on stake table endpoint
